drivers/regex/mlx5/mlx5.c

```diff
@@ -58,6 +63,116 @@ mlx5_regex_get_ib_device_match(const struct rte_pci_addr *addr)
 		rte_errno = ENOENT;
 	return ibv_match;
 }
+
+static int mlx5_regex_setup_dev(struct mlx5_regex_priv *priv)
+{
+	struct mlx5dv_pd mlx5_pd = {};
+	struct mlx5dv_obj dv_obj;
+	int ret;
+
+	ret = mlx5dv_devx_query_eqn(priv->ctx, 0, &priv->eqn);
+	if (ret || !priv->eqn) {
+		if (!ret)
+			ret = EINVAL;
+		return ret;
+	}
+
+	priv->pd = ibv_alloc_pd(priv->ctx);
+	if (!priv->pd)
+		return ENOMEM;
+	dv_obj.pd.in = priv->pd;
+	dv_obj.pd.out = &mlx5_pd;
+	mlx5dv_init_obj(&dv_obj, MLX5DV_OBJ_PD);
+	priv->pdn = mlx5_pd.pdn;
+
+	priv->uar = mlx5dv_devx_alloc_uar(priv->ctx, 0);
```

Comment:
If we have 1 uar it means we need to lock on every doorbell...
I discussed it with ori and agreed that we should have 1 UAR per QP.

Reply:
@yuvalavn ok. yes make sense. I will change it and move uar allocation to QP create flow.